pkg/api/types.go

```diff
@@ -530,6 +530,7 @@ type AgentPoolProfile struct {
 	AvailabilityZones                   []string             `json:"availabilityZones,omitempty"`
 	SinglePlacementGroup                *bool                `json:"singlePlacementGroup,omitempty"`
 	VnetCidrs                           []string             `json:"vnetCidrs,omitempty"`
+	NotPreserveNodesProperties          *bool                `json:"notPreserveNodesProperties,omitempty"`
```

Comment:
I thought the conclusion of the email discussion was to enable this by default with gradual rollout?

Reply:
If we do decide to keep the toggle `NotPreserveNodesProperties` sounds a bit redundant, I would vote for `PreserveNodesProperties` with a `True` default.

Reply:
changed